### PR TITLE
Add missing test cases

### PR DESCRIPTION
**Pull request by levanhieuhust**
**What is the purpose of the change**

- Update test case for **CreateTrade**, **SetItemString**, **GetRecipeHistory**, **ListSignUpByReferee**
- Write all test cases with table-driven test **CreateTrade**, **SetItemString**, **GetRecipeHistory**, **ListSignUpByReferee**

**Testing and Verifying**

- This change is a trivial rework/code cleanup without any test coverage

**Documentation and Release Note**

- Does this pull request introduces a new feature or user-facing behaviour changes? no
- How is the feature or change documented? not applicable

### CODE REVIEW

**Comment by faddat** [comment]
This is great.  I'm just working out how to use the GitHub mobile app but your code is wonderful and as soon as I find the approve button I will click it

**Comment by faddat** [comment]
This is great.  I'm just working out how to use the GitHub mobile app but your code is wonderful and as soon as I find the approve button I will click it

**Review by faddat — APPROVED** [review]
Thank you

**Comment by levanhieuhust** [comment]
Thank you, Sir !

File: x/pylons/keeper/msg_server_trade_test.go [hunk 1]
```diff
@@ -68,32 +18,33 @@ func (suite *IntegrationTestSuite) TestTradeMsgServerCreate1() {
...
-	// app.BankKeeper.SetParams(ctx, params)
 
 	wctx := sdk.WrapSDKContext(ctx)
 	srv := keeper.NewMsgServerImpl(k)
 
 	creator := types.GenTestBech32FromString("creator")
 
 	fooCoin := sdk.NewCoin("testPylons", sdk.OneInt())
-	var items []types.Item
+	// bondCoin := sdk.NewCoin("testPylons1", sdk.OneInt())
```
Comment (mustafapylons) [inline x/pylons/keeper/msg_server_trade_test.go:28]: is this code necessary, or dead code?

**Review by mustafapylons — COMMENTED** [review]
Few small issues, otherwise code is good.

File: x/pylons/keeper/msg_server_trade_test.go (hunk 1 above)
Comment (faddat) [reply x/pylons/keeper/msg_server_trade_test.go:28]: comments aren't code

File: x/pylons/keeper/msg_server_trade_test.go [hunk 2]
```diff
@@ -130,8 +111,23 @@ func (suite *IntegrationTestSuite) TestTradeMsgServerCreate1() {
...
 				ExtraInfo:   "extraInfo",
 			},
-			itemTradeable: false,
-			err:           sdkerrors.ErrInvalidRequest,
+			lockCoinfortrade: false,
+			itemTradeable:    false,
+			err:              sdkerrors.ErrInvalidRequest,
+		},
+		{
+			desc: "Item and Cookbook ID can not be trade",
```
[the remark this answers is not in the document]
Comment (levanhieuhust) [reply x/pylons/keeper/msg_server_trade_test.go]: mr. @faddat has fix this review point in https://github.com/Pylons-tech/pylons/pull/790/commits/44f5fe5c8debbd35616164be56221c8138420f42. Thank @faddat very much :D

**Review by mustafapylons — APPROVED** [review]
Looks good to me now.

File: x/pylons/keeper/msg_server_trade_test.go (hunk 1 above)
Comment (levanhieuhust) [reply x/pylons/keeper/msg_server_trade_test.go]: I have remove this line of code in https://github.com/Pylons-tech/pylons/pull/790/commits/7de5b33f5fcbc50325faaa6ca0a096ad79364181. Thank @mustafapylons and @faddat verymuch :+1:

**Comment by levanhieuhust** [comment]
Hello @mustafapylons and @faddat , I have update test for **grpc_query_get_recipe_history**, **grpc_query_list_referrals_by_address**, **msg_server_set_item_string**. Could you please check for these test file?

**Review by faddat — CHANGES_REQUESTED** [review]
Hey, I like the code, but can you please add godoc comments above functions?

Otherwise, full approve and thank you!

**Comment by levanhieuhust** [comment]
> Hey, I like the code, but can you please add godoc comments above functions?
> 
> Otherwise, full approve and thank you!



> Hey, I like the code, but can you please add godoc comments above functions?
> 
> Otherwise, full approve and thank you!

Thank Sir, I will add comment above function

**Comment by levanhieuhust** [comment]
> Hey, I like the code, but can you please add godoc comments above functions?
> 
> Otherwise, full approve and thank you!

I have add godoc comment for these test functions

**Review by faisalnaveedRNS — APPROVED** [review]

**Review by faddat — APPROVED** [review]

**Pull request merged**